scripts/make-g-cloud-7-live.py

```diff
@@ -0,0 +1,148 @@
+    elif domain.startswith('staging'):
+        return 'staging-assets.digitalmarketplace.service.gov.uk'
+    elif domain.startswith('www'):
+        return 'assets.digitalmarketplace.service.gov.uk'
+
+
+def get_live_asset_url(parsed_document, live_document_path):
+    domain = get_live_asset_domain(parsed_document['domain'])
+    return 'https://{}/{}'.format(domain, live_document_path)
+
+
+def make_draft_service_live(client, supplier, draft_service, framework_slug):
+    print("  > Migrating draft {} - {}".format(draft_service['id'], draft_service['serviceName']))
+    if DRY_RUN:
+        service_id = random.randint(1000, 10000)
+        print("    > generating random test service ID".format(service_id))
+    else:
+        services = client.publish_draft_service(draft_service['id'], 'make-g-cloud-7-live script')
+        service_id = services['services']['id']
+        print("    > draft service published - new service ID {}".format(service_id))
+
+    document_updates = {}
+    for document_key in DOCUMENT_KEYS:
+        if document_key not in draft_service:
+            print("    > Skipping {}".format(document_key))
+        else:
+            parsed_document = parse_document_url(draft_service[document_key], framework_slug)
+            assert_equal(str(parsed_document['supplier_id']), str(supplier['supplierId']))
+            assert_equal(str(parsed_document['draft_id']), str(draft_service['id']))
+
+            draft_document_path = get_draft_document_path(parsed_document, framework_slug)
+            live_document_path = get_live_document_path(parsed_document, framework_slug, service_id)
+            if not DRAFT_BUCKET.path_exists(draft_document_path):
+                raise ValueError(
+                    "Draft document {} does not exist in bucket {}".format(
+                        draft_document_path, DRAFT_BUCKET.bucket_name))
+            message_suffix = '{}:{} to {}:{}'.format(
+                DRAFT_BUCKET.bucket_name, draft_document_path,
+                DOCUMENTS_BUCKET.bucket_name, live_document_path
+            )
+            document_updates[document_key] = get_live_asset_url(parsed_document, live_document_path)
+
+            if DRY_RUN:
+                print("    > not copying {}".format(message_suffix))
+            else:
+                DOCUMENTS_BUCKET.bucket.copy_key(live_document_path, DRAFT_BUCKET.bucket_name, draft_document_path)
+                print("    > copied {}".format(message_suffix))
+
+    if DRY_RUN:
+        print("    > not updating document URLs {}".format(document_updates))
+    else:
+        client.update_service(service_id, document_updates, 'make-g-cloud-7-live script')
+        print("    > document URLs updated")
+
+
+if __name__ == '__main__':
+    arguments = docopt(__doc__)
+
+    client = DataAPIClient(arguments['<data_api_url>'], arguments['<data_api_token>'])
+    DRAFT_BUCKET = S3(arguments['<draft_bucket>'])
+    DOCUMENTS_BUCKET = S3(arguments['<documents_bucket>'])
```

Comment:
This is a minor point, but defining the global variables here sort of defeats the purpose of the `if __name__` condition. Functions rely on the variables being defined, so most of them won't work if the script is imported. Can we pass the buckets as arguments instead (similar to `client`)?